Extensions/AuthPropertiesTokenCacheHelper.cs

```diff
@@ -39,7 +39,7 @@ public class CookieTokenCacheProvider : ITokenCacheProvider
         /// <param name="signInScheme">Sign-in scheme</param>
         /// <returns>A token cache to use in the application</returns>
 
-        public TokenCache GetCache(HttpContext httpContext, AuthenticationProperties authenticationProperties, string signInScheme)
+        public TokenCache GetCache(HttpContext httpContext, ClaimsPrincipal claimsPrincipal, AuthenticationProperties authenticationProperties, string signInScheme)
```

Comment:
When redeeming the token httpContext.User does not have the claims yet (but they are in the AuthorizationCodeReceivedContext.Principal has them extracted from the IdToken)